### ISSUE TITLE
Specifying http.public.ssl_policy in the environment manifest creates CloudFormation with incorrect key of SSLPolicy.

### ISSUE THREAD

**Issue by Jones19** [aws/copilot-cli#4269]
Specifying the AWS SSL Security policy in the copilot environment manifest will  generate  CloudFormation in which the AWS::ElasticLoadBalancingV2::Listener resource  has a property key of "SSLPolicy" which will generate the following validation failure:

Resource template validation failed for resource HTTPSListener as the template has invalid properties. Please refer to the resource documentation to fix the template. Properties validation failed for resource HTTPSListener with message: #: extraneous key [SSLPolicy] is not permitted

The correct key value  should be "SslPolicy".

The CloudFormation does update the resource as expected, but will  detect drift in the AWS::ElasticLoadBalancingV2::Listener  due to the  validation issue.

**Comment by efekarakus** [comment]
Hi @Jones19 ! I am terribly sorry for the key error, looks like indeed the wrong key name was merged :( 

https://github.com/aws/copilot-cli/blob/21030979654a22e381284557cb1d5a6276939fcc/internal/pkg/template/templates/environment/cf.yml#L380-L382

vs. 

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticloadbalancingv2-listener.html#cfn-elasticloadbalancingv2-listener-sslpolicy

We'll create a fix and reflect on how we could have caught this issue earlier 💭

**Comment by efekarakus** [comment]
In the mean time, until we merge in a fix, since the `SslPolicy` doesn't require any replacement on update, I think it's safe to update the load balancer resource with the AWS CLI or the Console. Sorry for the inconvenience 🙇

**Comment by Jones19** [comment]
Thank you.

**Comment by efekarakus** [comment]
Hi @Jones19 !

We just tested the `ssl_policy` field with @KollaAdithya in `us-west-2`:
```yaml
http:
  public:
    ssl_policy: "ELBSecurityPolicy-TLS-1-0-2015-04"
```
and it actually deployed 🤯 !

So for some reason in `us-west-2`, which is where we originally tested the template, the deployment works although the capitalization does not conform to the documentation.

**In which region is your environment?** I wonder if `SSLPolicy` is valid only in particular regions and fails in others. We can try to reproduce in your region.

**Comment by Jones19** [comment]
Hi @efekarakus ,

So my experience was that the deployment DID work correctly in `us-east-2` as well. The CloudFormation did change the Security Policy on the ALB Listener as expected, but it also fails validation with the error. So I'm getting the correct result from the `copilot env deploy` but the CloudFormation events show the template validation failure.

**Comment by efekarakus** [comment]
oh, thanks for the clarification! How is this validation run? how/where does this error get surfaced?

**Comment by Jones19** [comment]
In the cfn stack for the specified environment, the Events column will show the `Resource Validation failed for resource HTTPSListener as the template has invalid properties...`

Also, detecting drift on the stack will show drift for the HTTPSListener. I ran the stack manually with the correct key name and there is no drift detected.

**Comment by Lou1415926** [comment]
Hello all! The fix is now released in [v1.25.0](https://github.com/aws/copilot-cli/releases/tag/v1.25.0)!
